src/Mutator/Arithmetic/Assignment.php

```diff
@@ -59,13 +60,11 @@ public static function getDefinition(): ?Definition
     }
 
     /**
-     * @param Node&Node\Expr\AssignOp $node
-     *
-     * @return Node\Expr\Assign
```

Comment:
Why not to preserve all return type hints wrapped with `Generator`? This will help PHPStan to control more mistakes: 

https://phpstan.org/r/531725c8-1222-4e75-ac55-964614565919

Reply:
I was under the impression that we don't really care about being more specific than `Generator<Node|Node[]>` here

Reply:
Why?

While we now use `$mutator->mutate($node);` only in one place (`NodeMutationGenerator`), it does not mean specific types will be useless in the future, especially when we extract mutators to a separate package.

In fact, they are useful even now:

```php
/**
 * @param Node $node
 * @return Generator<Node\Scalar\DNumber>
 */
public function mutate(Node $node): Generator
{
    {
        if ($node->value === 0.0) {
            if ($node->value === 0.0) {
                yield new Node\Scalar\DNumber(1.0);

                return;
            }
        }

        yield new Node\Scalar\DNumber(0.0);
    }
}
```

so, if in this mutator someone make a mistake, returning `LNumber` instead of `DNumber`, that would be reported by analyzer.

Anyway, replacing `Generator<type>` with `Generator` is like a decreasing type coverage, similar to replacing `int[]` with `array`.

Reply:
Fixed... that was tedious